### DISCUSSION TITLE
API response time not returned in the "minio_s3_requests_ttfb_seconds_distribution" metric

### DISCUSSION THREAD

**Discussion by bebianc** [minio/minio#19979] [Q&A]
As of "RELEASE.2023-09-30T07-02-29Z" the metric "**minio_s3_time_ttfb_seconds_distribution**" has changed to "**minio_s3_requests_ttfb_seconds_distribution**".
 However, the times of the "putobject", "listbuckets", "deleteobject" and other APIs are not returned when the "**curl -s http://localhost:9000/minio/v2/metrics/cluster**" or "**curl -s http://localhost:9000/minio/v2/metrics/node**" request is executed. 
It only returns the values of the "getobject" API, although in other key metrics, such as "minio_s3_requests_total", the values of the other APIs are successfully returned.

My question is whether "**RELEASE.2024-06-06T09:36:42Z**" really only returns the times from the "getobject" API or whether it is necessary to make some additional configuration so that the values from the other APIs are returned.

Example of the return of the metric "minio_s3_requests_total" from a request to the "/minio/v2/metrics/cluster" which shows the values of other APIs, in addition to "getobject":

:: HELP minio_s3_requests_total Total number of S3 requests
:: TYPE minio_s3_requests_total counter
minio_s3_requests_total{api="abortmultipartupload",server="127.0.0.1:9000"} 15
minio_s3_requests_total{api="completemultipartupload",server="127.0.0.1:9000"} 927
minio_s3_requests_total{api="deletemultipleobjects",server="127.0.0.1:9000"} 1
minio_s3_requests_total{api="deleteobject",server="127.0.0.1:9000"} 11
minio_s3_requests_total{api="getbucketversioning",server="127.0.0.1:9000"} 6
minio_s3_requests_total{api="**getobject**",server="127.0.0.1:9000"} 222

Return of the metric "minio_s3_requests_ttfb_seconds_distribution" from the same request "/minio/v2/metrics/cluster" which shows the return of only the API "**getobject**":

:: HELP minio_s3_requests_ttfb_seconds_distribution Distribution of time to first byte across API calls
:: TYPE minio_s3_requests_ttfb_seconds_distribution gauge
minio_s3_requests_ttfb_seconds_distribution{api="GetObject",le="+Inf",server="127.0.0.1:9000"} 0
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="0.050",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="0.100",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="0.250",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="0.500",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="1.000",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="10.000",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="2.500",server="127.0.0.1:9000"} 222
minio_s3_requests_ttfb_seconds_distribution{api="getobject",le="5.000",server="127.0.0.1:9000"} 222

Any help is welcome.
Thanks from Brazil :D

**Comment by harshavardhana** [comment]
ttfb is not response time @bebianc - it is time to first byte. There is no histogram for overall response time.

**Comment by bebianc** [comment]
Thanks for supporting me @harshavardhana
